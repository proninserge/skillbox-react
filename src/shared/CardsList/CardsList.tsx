import { Card } from './Card/Card';
import styles from './cardslist.less';
import withKey from '@utils/react/withKey';
import {getPosts} from '@utils/utils';
import { RootState } from '@store/reducer/root-reducer';
import {IPostsData} from '@utils/react/hooks/usePostsData';
import { useSelector } from 'react-redux';

export function CardsList() {

    const posts = useSelector<RootState, IPostsData[]>(state => state.posts);

    return (
        <>
            {
                posts.length !== 0 ?
                <ul className={styles.cardsList}>
                    {getPosts(posts).map(withKey('id')(Card))}
                </ul> : 
                <h2 className={styles.preload}>Nothing here just yet. Please sign in</h2>
            }
        </>
    );
}