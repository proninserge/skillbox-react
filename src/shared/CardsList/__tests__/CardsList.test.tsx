import {shallow, configure} from 'enzyme';
import {CardsList} from '../CardsList';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('CardsList', () => {
    const wrapper = shallow(<CardsList />);

    test('should render', () => {
        expect(wrapper).toBeDefined();
        expect(wrapper.find('ul.cardsList').isEmptyRender()).toBeFalsy();
    });

    test ('should render snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

});