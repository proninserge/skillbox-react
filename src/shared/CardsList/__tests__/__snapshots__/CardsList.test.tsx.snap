// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardsList should render snapshot 1`] = `
<ul
  className="cardsList"
>
  <Card />
</ul>
`;
